manager_dashboard/manager_dashboard/js/usergroup-management.js

```diff
@@ -0,0 +1,115 @@
+$("#add-usergroup-modal").on('show.bs.modal', function() {
+  $('#usergroup-name').val('');
+  $('#usergroup-description').val('');
+})
+
+function submitNewUserGroup() {
+  var nameInput = $('#usergroup-name');
+  var name = (nameInput.val() || '').trim();
+  var descriptionInput = $('#usergroup-description');
+  var description = (descriptionInput.val() || '').trim();
+
+  if (name && name.length <= 50) {
+    $('#add-usergroup-modal-loading-message-container').removeClass('hidden');
+    nameInput.attr('disabled', true);
+    descriptionInput.attr('disabled', true);
+
+    firebase.database().ref('v2/userGroups/' + name).once("value", function(snapshot) {
+      if (snapshot.exists()) {
+        $('#add-usergroup-modal-loading-message-container').addClass('hidden');
+        nameInput.attr('disabled', false);
+        descriptionInput.attr('disabled', false);
+        alert('A group with this name already exists');
+      } else {
+        var updates = {};
+        updates['/v2/userGroups/' + name] = {
+          userGroupName: name,
+          description,
+        };
+        firebase.database().ref().update(updates, function(error) {
+          $('#add-usergroup-modal-loading-message-container').addClass('hidden');
+          nameInput.attr('disabled', false);
+          descriptionInput.attr('disabled', false);
+
+          if (error) {
+            alert('Failed to add new user');
+            console.info(error);
+          } else {
+            alert('Successfully added new user group');
+            window.location.reload();
+          }
+        });
+      }
+    });
+  } else {
+    nameInput.attr('disabled', false);
+    descriptionInput.attr('disabled', false);
+    $('#add-usergroup-modal-loading-message-container').addClass('hidden');
+    alert('User group name is required and should be less than 50 characters');
```

Comment:
Same comment as above, you've limited this to 30 elsewhere.

Reply:
Done!